presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -80,6 +83,7 @@
     private final AtomicReference<Integer> networkTimeoutMillis = new AtomicReference<>(Ints.saturatedCast(MINUTES.toMillis(2)));
     private final AtomicReference<ServerInfo> serverInfo = new AtomicReference<>();
     private final AtomicLong nextStatementId = new AtomicLong(1);
+    private final AtomicInteger statementDepth = new AtomicInteger(0);
```

Comment:
I think we should move this state into `PrestoStatement`.  A connection can create multiple statements, all of which can be executed independently and in different threads.  On the other hand, there is always just one statement per result set, so the state really should apply to just the statement.

Reply:
Yeah that's what I was afraid of. If we put the depth inside the statement, wouldn't we lose that depth if the QueryInterceptor creates another Statement? I can test it to make sure.